rules/call_peak.smk

```diff
@@ -1,32 +1,26 @@
-def get_genrich_replicates(wildcards):
-    assembly, sample_condition = "-".join(wildcards.fname.split('-')[:-1]), wildcards.fname.split('-')[-1]
-    if not 'condition' in samples.columns or config.get('combine_replicates', '') == 'merge' \
-    or (sample_condition in samples.index and not sample_condition in samples['condition'].values):
-        return expand(f"{{dedup_dir}}/{wildcards.fname}.sambamba-queryname.bam", **config)
-    else:
-        return expand([f"{{dedup_dir}}/{assembly}-{replicate}.sambamba-queryname.bam"
-        for replicate in samples[(samples['assembly'] == assembly) & (samples['condition'] == sample_condition)].index], **config)
-
-
 rule genrich_pileup:
     """
-    Generate the pileup. We do this separately from peak-calling since these two processes have a very different
+    Generate the pileup. For >1 sample, combined with Fisher's method, we use genrich_pileup_fisher 
+    
+    We do this separately from peak-calling since these two processes have a very different
     computational footprint.
     """
     input:
-        get_genrich_replicates
+        expand("{dedup_dir}/{{assembly}}-{{sample}}.sambamba-queryname.bam", **config)
     output:
-        bedgraphish=expand("{result_dir}/genrich/{{fname}}.bdgish", **config),
-        log=expand("{result_dir}/genrich/{{fname}}.log", **config)
+        bedgraphish=expand("{result_dir}/genrich/{{assembly}}-{{sample}}.bdgish", **config),
+        log=expand("{result_dir}/genrich/{{assembly}}-{{sample}}.log", **config)
     log:
-        expand("{log_dir}/genrich_pileup/{{fname}}_pileup.log", **config)
+        expand("{log_dir}/genrich_pileup/{{assembly}}-{{sample}}_pileup.log", **config)
     benchmark:
-        expand("{benchmark_dir}/genrich_pileup/{{fname}}.benchmark.txt", **config)[0]
+        expand("{benchmark_dir}/genrich_pileup/{{assembly}}-{{sample}}.benchmark.txt", **config)[0]
     conda:
         "../envs/genrich.yaml"
     params:
         config['peak_caller'].get('genrich', " ")  # TODO: move this to config.schema.yaml
-    threads: 15  # TODO: genrich uses lots of ram. Get the number from benchmark, instead of doing it through threads
+    threads: 15
+    resources:
+        mem_gb=7
```

Comment:
It uses much, much more than 7. Maybe change it into 20 or so. Then you can also set 1 thread